overlord/snapstate/handlers.go

```diff
@@ -2973,10 +2989,67 @@ func (m *SnapManager) doToggleSnapFlags(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func installModeDisabledSystemServices(snapst *SnapState, currentInfo *snap.Info, prevCurrentSvcs map[string]bool) (svcsToDisable []string) {
+	enabledByHookSvcs := map[string]bool{}
+	for _, svcName := range snapst.ServicesEnabledByHooks {
+		enabledByHookSvcs[svcName] = true
+	}
+	for _, svc := range currentInfo.Services() {
+		if svc.DaemonScope != snap.SystemDaemon {
+			continue
+		}
+
+		if svc.InstallMode == "disable" && !enabledByHookSvcs[svc.Name] {
+			if !prevCurrentSvcs[svc.Name] {
+				svcsToDisable = append(svcsToDisable, svc.Name)
+			}
+		}
+	}
+	return svcsToDisable
+}
+
+// installModeDisabledUserServices returns a map of currently active users
+// with user services that have been marked for 'install-mode: disable', which
+// were not already disabled for each of the active users.
+// The reason we are doing this only for users currently logged in, is because we
+// do a best-effort handling of user sewrvices - we can only query the user service
+// agent for users that have it running.
+func installModeDisabledUserServices(snapst *SnapState, currentInfo *snap.Info, prevCurrentSvcs map[string]bool) (map[int][]string, error) {
+	availableUids, err := clientutil.AvailableUserSessions()
+	if err != nil {
+		return nil, err
+	}
+
+	enabledByHookSvcs := make(map[int]map[string]bool)
+	for uid, svcs := range snapst.UserServicesEnabledByHooks {
+		enabledByHookSvcs[uid] = make(map[string]bool)
+		for _, svcName := range svcs {
+			enabledByHookSvcs[uid][svcName] = true
+		}
+	}
+
+	svcsToDisable := make(map[int][]string)
+	for _, svc := range currentInfo.Services() {
+		if svc.DaemonScope != snap.UserDaemon {
+			continue
+		}
+
+		if svc.InstallMode == "disable" && !prevCurrentSvcs[svc.Name] {
+			// determine if it was enabled for the any of the users
+			for _, uid := range availableUids {
+				if len(enabledByHookSvcs[uid]) == 0 || !enabledByHookSvcs[uid][svc.Name] {
+					svcsToDisable[uid] = append(svcsToDisable[uid], svc.Name)
+				}
+			}
+		}
+	}
+	return svcsToDisable, nil
+}
+
 // installModeDisabledServices returns what services with
 // "install-mode: disabled" should be disabled. Only services
 // seen for the first time are considered.
-func installModeDisabledServices(st *state.State, snapst *SnapState, currentInfo *snap.Info) (svcsToDisable []string, err error) {
+func installModeDisabledServices(st *state.State, snapst *SnapState, currentInfo *snap.Info) (sysSvcsToDisable []string, usrSvcsToDisable map[int][]string, err error) {
 	enabledByHookSvcs := map[string]bool{}
 	for _, svcName := range snapst.ServicesEnabledByHooks {
 		enabledByHookSvcs[svcName] = true
```

Comment:
is this map used in the new code?

Reply:
You are absolutely right - it was a leftover from moving the code!